### ISSUE TITLE
Initialize `out_layer` only when it's needed

### ISSUE THREAD

**Issue by IaroslavElistratov** [themis-ai/capsa#95]
Currently the `out_layer` is [always initialized](https://github.com/themis-ai/capsa/blob/2fc6e89da56997bf962a07c1bf268b9523171327/capsa/base_wrapper.py#L67) in the `BaseWrapper`. Even for the metric wrappers that don't use it (e.g., `EnsembleWrapper`, `DropoutWrapper`). Which results in some warning messages being printed in the beginning of training and when saving these models (please see below).

```
WARNING:tensorflow:Skipping full serialization of Keras layer <keras.layers.core.dense.Dense object at 0x7ff230086ac0>, because it is not built.
```
